pkg/sql/sqlbase/table.go

```diff
@@ -1579,27 +1615,59 @@ func MarshalColumnValue(col ColumnDescriptor, val parser.Datum) (roachpb.Value,
 		val.ResolvedType(), col.Type.SemanticType, col.Name)
 }
 
+func encodeArrayHeader(h arrayHeader, buf []byte) ([]byte, error) {
+	// The byte we append here is formatted as follows:
+	// * The high 4 bits encode the number of dimensions in the array.
```

Comment:
I just realized something, sorry for being late about it: when designing encodings it's rather common+desirable to place the flags that are more likely to be zero in the more significant bits. The reason is that for when a human looks at a value it is good that the most common values also be the smaller in numeric value.

Hence my concrete suggestion to place the dimensions in the LSB and the null bit above that.